archive_viewer/table_models/curve_model.py

```diff
@@ -211,3 +257,14 @@ def curve_at_index(self, index: QModelIndex) -> ArchivePlotCurveItem:
             The requested curve.
         """
         return self._plot.curveAtIndex(index)
+
+    @Slot(object)
+    def remove_curve(self, curve: BasePlotCurveItem) -> None:
+        """Necessary specifically for when an axis is deleted
+        To properly delete all of its connected curves
+
```

Comment:
this is very nit-picky of me but can you add, 

Parameters
----------

to the docstring before the 'curve: BasePlotCurveItem'